chia/data_layer/data_store.py

```diff
@@ -1203,6 +1203,9 @@ async def build_ancestor_table_for_latest_root(self, tree_id: bytes32) -> None:
                 if node.hash not in known_hashes:
                     await self._insert_ancestor_table(node.left_hash, node.right_hash, tree_id, root.generation)
 
+            # Clean up intermediary hashes that are no longer needed, since every hash we need should be in ancestors.
+            await writer.execute("DELETE FROM node WHERE hash NOT IN (SELECT hash FROM ancestors)")
```

Comment:
Is this a scalable query?  I don't know how it gets executed but if we have lots of ancestors it might get big.  I'm wondering if there's some way to leverage foreign keys to tell SQLite what to delete maybe?

Reply:
I didn't get to test it for very large data sets (only what's included in the tests). but my reasoning is that this happens only once (when a new root gets confirmed on chain), so as long as it's not _completely_ terrible, we should still be fine given it's executed rarely.

Reply:
actually, now that I look at it, we might have to run it one more time at startup - to have backwards compatibility for stores created before this PR.

Reply:
Ok, we can remember this anyways f we start hearing about relevant performance issues.  Feel free to resolve this conversation after reading.